test/studies/shootout/binary-trees/binarytrees-pool.chpl

```diff
@@ -8,31 +8,35 @@
 
 use Allocators;
 
-config const n = 10,         // the maximum tree depth
-             globalPoolSize = (2 ** 31)-1,
-             localPoolSize = globalPoolSize;
-
-var globalPool = new bumpPtrMemPool(globalPoolSize, alignment=0);
+config const n = 10;                       // the maximum tree depth
 
 proc main() {
   const minDepth = 4,                      // the shallowest tree
         maxDepth = max(minDepth + 2, n),   // the deepest normal tree
         strDepth = maxDepth + 1,           // the depth of the "stretch" tree
-        depths = minDepth..maxDepth by 2;  // the range of depths to create
+        depths = minDepth..maxDepth by 2,  // the range of depths to create
+        nodeSize = 24;                     // the approximate size of a node
   var stats: [depths] (int,int);           // stores statistics for the trees
+  const poolSize =
+    (((2**(maxDepth+2))-1)*nodeSize)*3;    // how much each pool allocates
+                                           // based on the deepest tree,
+                                           // with a factor of 3 for safety
+  var globalPool = new bumpPtrMemPool(poolSize, alignment=0);
 
   //
   // Create the short-lived "stretch" tree, checksum it, and print its stats.
   //
   {
-    const strTree = newWithAllocator(globalPool, unmanaged Tree, strDepth, globalPool);
+    const strTree = newWithAllocator(globalPool,
+                                      unmanaged Tree, strDepth, globalPool);
```

Comment:
If you were to rename `globalPool` to simply `pool`, these wrapping lines would fit on one line each, which seems attractive.  If it felt asymmetric to have `pool` and `localPool`, you could call `localPool` `myPool` or `taskPool` (or even `pool`, though… maybe that's sketchy style-wise…?  though it would also probably gzip really nicely since all these expressions would be the same :D ).

Reply:
I actually don't mind naming both `pool` and letting shadowing deal with it